extension/src/cli/dvc/reader.ts

```diff
@@ -156,11 +156,16 @@ export class DvcReader extends DvcCli {
     defaultValue: string,
     ...args: Args
   ): Promise<T> {
-    const output =
-      (await retry(
-        () => this.executeDvcProcess(cwd, ...args),
-        args.join(' ')
-      )) || defaultValue
+    let output
+    try {
+      output = await this.executeDvcProcess(cwd, ...args)
+    } catch (error: unknown) {
+      const errorMessage = (error as Error).message
+      Logger.error(`${args} failed with ${errorMessage} retrying...`)
+    }
+
+    output = output || defaultValue
```

Comment:
[F] I'll have to change all of the types to include an optional error which I can then parse/collect and display in the UI.